json/src/main/java/com/basistech/rosette/apimodel/jackson/recordsimilaritydeserializers/RecordSimilarityResultDeserializer.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2024 Basis Technology Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.basistech.rosette.apimodel.jackson.recordsimilaritydeserializers;
+
+import java.io.IOException;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+import com.basistech.rosette.apimodel.recordsimilarity.RecordSimilarityExplainInfo;
+import com.basistech.rosette.apimodel.recordsimilarity.RecordSimilarityFieldInfo;
+import com.basistech.rosette.apimodel.recordsimilarity.RecordSimilarityResult;
+
+public class RecordSimilarityResultDeserializer extends StdDeserializer<RecordSimilarityResult> {
+
+    public RecordSimilarityResultDeserializer() {
+        super(RecordSimilarityResult.class);
+    }
+
+    @Override
+    public RecordSimilarityResult deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
+        final JsonNode node = jsonParser.getCodec().readTree(jsonParser);
+        final Map<String, RecordSimilarityFieldInfo> fields = node.get("fields").traverse(jsonParser.getCodec()).readValueAs(new TypeReference<Map<String, RecordSimilarityFieldInfo>>() { });
```

Comment:
result json does NOT have fields. We ha to fix this. Thats why my IT fails.

Reply:
im not sure i understand what you mean. in `RecordSimilarityRequestTest` it should be testing that it serializes and deserializes correctly, including having the "fields" field

Reply:
This method deserializes json String to a RecordSimialrityResult object. The problem is that we expect that the json text has a "fields" field (at line 41) which the result from the query doesnt have. For example here is an example result:
`"{"extendedInformation":null,"results":[{"score":0.9155403504970899,"left":{"name1":"John Malecki"},"right":{"name1":"Malecki John"}}],"errorMessage":""}"`
Yesterday i spoke with @fhasanaj about this and she said that this method is not going to be used. 
But I disaggrie: We (currently) are not able to convert the result json to a RecordSimilarityRespone object. Think to the java binding examples. I think in that code we want to deserialize back the json back to a previously mentioned object.